enterprise/frontend/src/embedding-sdk/cli/steps/setup-permission.ts

```diff
@@ -0,0 +1,156 @@
+import chalk from "chalk";
+
+import { SANDBOXED_GROUP_NAMES } from "../constants/config";
+import { NOT_ENOUGH_TENANCY_COLUMN_ROWS } from "../constants/messages";
+import type { CliStepMethod } from "../types/cli";
+import { getCollectionPermissions } from "../utils/get-collection-permissions";
+import { getPermissionsForGroups } from "../utils/get-permission-groups";
+import { getTenancyIsolationSandboxes } from "../utils/get-tenancy-isolation-sandboxes";
+import {
+  cliError,
+  propagateErrorResponse,
+} from "../utils/propagate-error-response";
+import { sampleTenantIdsFromTables } from "../utils/sample-tenancy-column-values";
+
+export const setupPermissions: CliStepMethod = async state => {
+  const { cookie = "", instanceUrl = "", tenancyColumnNames = {} } = state;
+
+  let res;
+  const collectionIds: number[] = [];
+
+  // Create new collections sequentially
+  try {
+    for (const groupName of SANDBOXED_GROUP_NAMES) {
+      res = await fetch(`${instanceUrl}/api/collection`, {
+        method: "POST",
+        headers: { "content-type": "application/json", cookie },
+        body: JSON.stringify({
+          parent_id: null,
+          authority_level: null,
+          color: "#509EE3",
+          description: null,
+          name: groupName,
+        }),
+      });
+
+      await propagateErrorResponse(res);
+
+      const { id: collectionId } = (await res.json()) as { id: number };
+      collectionIds.push(collectionId);
+    }
+  } catch (error) {
+    const message = `Failed to create customer collections`;
+
+    return [cliError(message, error), state];
+  }
+
+  // Example: { "Customer A": [3], "Customer B": [4], "Customer C": [5] }
+  const jwtGroupMappings: Record<string, number[]> = {};
+
+  try {
+    // Create new permission groups
+    for (const groupName of SANDBOXED_GROUP_NAMES) {
+      res = await fetch(`${instanceUrl}/api/permissions/group`, {
+        method: "POST",
+        headers: { "content-type": "application/json", cookie },
+        body: JSON.stringify({ name: groupName }),
+      });
+
+      await propagateErrorResponse(res);
+
+      const { id: groupId } = (await res.json()) as { id: number };
+
+      jwtGroupMappings[groupName] = [groupId];
+    }
+
+    // Update the JWT group mappings.
```

Comment:
This allows the mock users to have the correct permission groups, for sandboxing demo purposes.